### PR TITLE
 Load annotation upvote on Viewer appearance + Some GA lost commits

### LINKED ISSUE #174: Load annotation upvote on Viewer appearance
Download upvote lazily for an annotation that is about to appear.

Major improvement: with current backend structure, upvote instance handling will be much cleaner.

### PR DESCRIPTION
**Pull request by glowka**
Resolves #174

### CODE REVIEW

File: src/components/viewer/Upvote.tsx [hunk 1]
```diff
@@ -14,19 +14,27 @@ import ppGA from 'pp-ga';
...
   annotation: AnnotationAPIModel;
+  upvote: AnnotationUpvoteAPIModel;
 
+  fetchUpvote: (url: string) => Promise<object>;
   deleteUpvote: (instance: AnnotationUpvoteAPIModel) => Promise<object>;
   createUpvote: (instance: AnnotationUpvoteAPICreateModel) => Promise<object>;
 }
 
 interface IUpvoteState {
-  initialView: boolean; // used to determine whether edit/delete buttons should be visible
```
Comment (kaglowka) [inline src/components/viewer/Upvote.tsx]: This field was used by edit/delete buttons to determine whether they are still visible, I suppose that's an unintended change :)

Comment (kaglowka) [reply src/components/viewer/Upvote.tsx]: (they disappear after some time when the mouse is not over an annotation any more)

Comment (glowka) [reply src/components/viewer/Upvote.tsx]: It was obsolete code after file splitting, this field is used elsewhere :P

File: src/components/viewer/Upvote.tsx [hunk 2]
```diff
@@ -35,23 +43,22 @@ export default class Upvote extends React.Component<Partial<IUpvoteProps>, Parti
 
   constructor(props: IUpvoteProps) {
     super(props);
+
+    if (props.annotation.relationships.annotationUpvote.data && !props.upvote) {
+      props.fetchUpvote(props.annotation.relationships.annotationUpvote.links.related).then(
```
Comment (kaglowka) [inline src/components/viewer/Upvote.tsx]: So slick :+1:  :)

**Pull request merged**